tests/e2e/fixtures/storage_efs_dependencies.py

```diff
@@ -229,23 +247,119 @@ def on_create():
+        efs_permission["spec"]["template"]["spec"]["volumes"][0][
+            "persistentVolumeClaim"
+        ]["claimName"] = claim_name
+        write_cfg(efs_permission, efs_permissions_filepath)
+
         kubectl_apply(efs_sc_filepath)
         kubectl_apply(efs_pv_filepath)
         kubectl_apply(efs_pvc_filepath)
+        kubectl_apply(efs_permissions_filepath)
 
         efs_claim["claim_name"] = claim_name
 
     def on_delete():
         kubectl_delete(efs_pvc_filepath)
         kubectl_delete(efs_pv_filepath)
         kubectl_delete(efs_sc_filepath)
+        kubectl_delete(efs_permissions_filepath)
 
     return configure_resource_fixture(
         metadata, request, efs_claim, "efs_claim", on_create, on_delete
     )
+
+
+@pytest.fixture(scope="class")
+def dynamic_provisioning(metadata, region, request, cluster):
+    claim_name = rand_name("efs-claim-auto-dyn-")
+    efs_pvc_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/dynamic-provisioning/pvc.yaml"
+    )
+    efs_sc_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/dynamic-provisioning/sc.yaml"
+    )
+    efs_permissions_filepath = (
+        "../../docs/deployment/add-ons/storage/notebook-sample/set-permission-job.yaml"
+    )
+    efs_auto_script_filepath = (
+        "../../docs/deployment/add-ons/storage/efs/auto-efs-setup.py"
+    )
+    efs_claim_dyn = {}
+    efs_client = get_efs_client(region)
+
+    def on_create():
+        # Run the automated script to create the EFS Filesystem and the SC
+        efs_auto_script_absolute_filepath = os.path.join(
+            os.path.abspath(sys.path[0]), "../" + efs_auto_script_filepath
+        )
+        st = os.stat(efs_auto_script_filepath)
+        os.chmod(efs_auto_script_filepath, st.st_mode | stat.S_IEXEC)
+        subprocess.call(
+            [
+                "python",
+                efs_auto_script_absolute_filepath,
+                "--region",
+                region,
+                "--cluster",
+                cluster,
+                "--directory",
+                "../../docs/deployment/add-ons/storage/efs/",
+                "--efs_file_system_name",
+                claim_name,
+            ]
+        )
+
+        file_system_id = get_file_system_id_from_name(efs_client, claim_name)
+
+        # PVC creation is not a part of the script
+        # Update the values in the pvc.yaml file
+        efs_pvc = load_cfg(efs_pvc_filepath)
+        efs_pvc["metadata"]["namespace"] = DEFAULT_USER_NAMESPACE
+        efs_pvc["metadata"]["name"] = claim_name
+        write_cfg(efs_pvc, efs_pvc_filepath)
+
+        kubectl_apply(efs_pvc_filepath)
+
+        efs_claim_dyn["efs_claim_dyn"] = claim_name
+        efs_claim_dyn["file_system_id"] = file_system_id
+
+    def on_delete():
+        kubectl_delete(efs_pvc_filepath)
+        kubectl_delete(efs_sc_filepath)
+
+        # Get FileSystem_ID
+        efs_claim_dyn = metadata.get("efs_claim_dyn") or efs_claim_dyn
```

Comment:
do you think a part of this code can become a `efs_cleanup` script? so the docs can reuse it in the cleanup steps

Reply:
yes, will do.

Reply:
Actually this needs a little more work, will create a separate PR for this.